.github/workflows/vvv-provisioning.yml

```diff
@@ -0,0 +1,91 @@
+# This is a basic workflow to help you get started with Actions
+
+name: VVV Provisioning
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the develop branch
+on:
+  push:
+    branches: [ develop ]
+  pull_request:
+    branches: [ develop, stable ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    name: Run Provisioner
+    # The type of runner that the job will run on
+    runs-on: ubuntu-18.04
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      # Make Symlinks
+      - name: Create Vagrant Like Environment
+        run: |
+          # uninstall pre installed packages (to test if utilities work)
+          sudo apt-get -q --autoremove --purge remove php*
+          sudo apt-get -q autoclean
+
+          # create vagrant user
+          sudo groupadd -g 2000 vagrant
+          sudo useradd -u 2000 -g vagrant -m vagrant
+
+          # Install bindfs, which would allow us to "mount" folders with specific permissions
+          # similar to vagrant shares
+          sudo apt-get update
+          sudo apt-get install bindfs -y
+
+          # vbox_mount function to sumulate synced folders
+          function vbox_mount() {
+            sudo mkdir -p "$2"
+            sudo bindfs --chown-ignore --chgrp-ignore --chmod-ignore -u "$3" -g "$4" "$1" "$2"
+          }
+
+          # create srv folder
+          sudo -u "vagrant" mkdir -p "/srv"
+
+          # copy files provided by vagrant
+          sudo cp -f "$GITHUB_WORKSPACE/config/default-config.yml" "$GITHUB_WORKSPACE/config/config.yml"
```

Comment:
Is this necessary? The vagrantfile copies `config/config.yml` to `/vagrant/config.yml` and that's what the provisioners work off of, in this case could we not copy it directly there? Or even a special version with a cut down site list?

Reply:
We do not run vagrantfile here. Therefore this is required unless we find a way to run VirtualBox in a container.

Reply:
I know, but since there is no host of any kind, why copy the default config to `config/config.yml` then copy that file again to `/vagrant/config.yml`? The provisioners never load `config/config.yml`, that's the vagrantfiles job, but since there is no vagrant file, we don't need it. Stop emulating the actions the host takes, and instead emulate the actions inside the environment.

Reply:
https://github.com/Varying-Vagrant-Vagrants/VVV/blob/develop/provision/provision.sh#L44-L51

You can see that if the main provisioner clear whatever is on /vagrant/config.yml, and copies the file from the config/config.yml location. This file does not exist on the repo, and does not exist until created via Vagrantfile. If we don't do that step manually at any point, then we cannot run the provisioners.

One thing I thought tho, is to actually move the code that copies the file from default-config.yml to config.yml to the main provisioner, or something that runs before the main provisioner, as it's something still tying us to Vagrant.

I'm honestly trying to do at least as possible to get the provisioners to run on the current state. This clearly also exposed some things that could be moved to the provisioners to simplify the Vagrantfile further, and remove dependency to it.

Reply:
Ah I see, lets resolve this then